stac_fastapi/pgstac/stac_fastapi/pgstac/types/search.py

```diff
@@ -14,8 +14,30 @@ class PgstacSearch(BaseSearchPostRequest):
     """
 
     datetime: Optional[str] = None
+    conf: Optional[Dict] = None
 
     @validator("datetime")
     def validate_datetime(cls, v):
         """Pgstac does not require the base validator for datetime."""
         return v
+
+    @validator("filter_lang", pre=False, check_fields=False, always=True)
+    def validate_query_uses_cql(cls, v, values):
+        """If using query syntax, forces cql-json."""
+        retval = v
+        if values.get("query", None) is not None:
+            retval = "cql-json"
+        if values.get("collections", None) is not None:
+            retval = "cql-json"
+        if values.get("ids", None) is not None:
+            retval = "cql-json"
+        if values.get("datetime", None) is not None:
+            retval = "cql-json"
+        if values.get("bbox", None) is not None:
+            retval = "cql-json"
+        if v == "cql2-json" and retval == "cql-json":
+            raise ValueError(
+                "query, collections, ids, datetime, and bbox"
+                "parameters are not available in cql2-json"
+            )
+        return retval
```

Comment:
`cql-text` and `cql-json` share the parameters listed here, but if `filter_lang` is "cql-text", it will be transformed to "cql-json". Is that the intention?

We might be able to condense this and avoid some duplicate code with the following snippet. It does change the logic slightly, though:

* It will throw if any CQL parameter is provided, even if its value is `null`. e.g. `{"filter-lang": "cql2-json", "query": null}` will throw the ValueError. I feel like that is still correct.
* It will only default to `cql-json` if `filter-lang` isn't specified (i.e. it will preserve `cql-text`)

```
cql_params = ("query", "collections", "ids", "datetime", "bbox")
if any(cql_param in values for cql_param in cql_params):
    if v == FilterLang.cql2_json:
        raise ValueError(f"{', '.join(invalid_cql2_params[:-1])} and {invalid_cql2_params[-1]} parameters are not available in cql2-json")
    else:
        return v or FilterLang.cql_json  # if `filter-lang` isn't specified and CQL parameters are provided, default to `cql-json`
return v
```

Reply:
PGStac only accepts json, so in the case of a GET request, that request is just modified into CQL json.  The ultimate plan (which I will start working on this week, but probably won't finish before the holidays) for stac-fastapi-pgstac will be to add a cql2-json output to pygeofilter and to use pygeofilter to translate cql-text or cql2-text into cql2-json and then send that to pgstac.